{"ast":null,"code":"var _jsxFileName = \"C:\\\\Nodejs\\\\ecomm\\\\client\\\\src\\\\components\\\\error-boundary\\\\ErrorBoundary.js\";\nimport React from 'react';\n\nclass ErrorBoundary extends React.Component {\n  constructor() {\n    super();\n    this.state = {\n      hasErrored: false\n    };\n  }\n\n  static getDerivedStateFromError(error)\n  /* This static method catches any error that gets thrown in any of the children \r\n  of this error boundary component*/\n  {\n    //process the error\n    return {\n      hasErrored: true\n    };\n  }\n\n  componentDidCatch(error, info) {\n    console.log(error);\n  }\n\n  render() {\n    if (this.state.hasErrored) {\n      return /*#__PURE__*/React.createElement(\"div\", {\n        __self: this,\n        __source: {\n          fileName: _jsxFileName,\n          lineNumber: 23,\n          columnNumber: 20\n        }\n      }, \"Something went wrong!!\");\n    }\n\n    return this.props.children;\n  }\n\n}\n\nexport default ErrorBoundary;","map":{"version":3,"sources":["C:/Nodejs/ecomm/client/src/components/error-boundary/ErrorBoundary.js"],"names":["React","ErrorBoundary","Component","constructor","state","hasErrored","getDerivedStateFromError","error","componentDidCatch","info","console","log","render","props","children"],"mappings":";AAAA,OAAOA,KAAP,MAAkB,OAAlB;;AAEA,MAAMC,aAAN,SAA4BD,KAAK,CAACE,SAAlC,CAA2C;AACvCC,EAAAA,WAAW,GAAE;AACT;AACA,SAAKC,KAAL,GAAW;AACPC,MAAAA,UAAU,EAAE;AADL,KAAX;AAGH;;AACD,SAAOC,wBAAP,CAAgCC,KAAhC;AAAsC;;AAEtC;AACI;AACA,WAAO;AAAEF,MAAAA,UAAU,EAAE;AAAd,KAAP;AACH;;AAEDG,EAAAA,iBAAiB,CAACD,KAAD,EAAQE,IAAR,EAAa;AAC1BC,IAAAA,OAAO,CAACC,GAAR,CAAYJ,KAAZ;AACH;;AAEDK,EAAAA,MAAM,GAAE;AACJ,QAAG,KAAKR,KAAL,CAAWC,UAAd,EAAyB;AACrB,0BAAO;AAAA;AAAA;AAAA;AAAA;AAAA;AAAA;AAAA,kCAAP;AACH;;AACD,WAAO,KAAKQ,KAAL,CAAWC,QAAlB;AACH;;AAvBsC;;AA2B3C,eAAeb,aAAf","sourcesContent":["import React from 'react'\r\n\r\nclass ErrorBoundary extends React.Component{\r\n    constructor(){\r\n        super()\r\n        this.state={\r\n            hasErrored: false\r\n        }\r\n    }\r\n    static getDerivedStateFromError(error)/* This static method catches any error that gets thrown in any of the children \r\n    of this error boundary component*/\r\n    {\r\n        //process the error\r\n        return { hasErrored: true}\r\n    }\r\n\r\n    componentDidCatch(error, info){\r\n        console.log(error)\r\n    }\r\n\r\n    render(){\r\n        if(this.state.hasErrored){\r\n            return <div>Something went wrong!!</div>\r\n        }\r\n        return this.props.children \r\n    }\r\n}\r\n\r\n\r\nexport default ErrorBoundary"]},"metadata":{},"sourceType":"module"}